dvc/remote/http.py

```diff
@@ -120,18 +121,37 @@ def _etag(self, url):
 
         return etag
 
-    def _download_to(self, url, file, callback=None):
+    def _download_to(self, url, target_file, callback=None):
         r = self._request('GET', url, stream=True)
 
-        with open(file, 'wb') as fd:
-            bytes_transfered = 0
-
-            for chunk in r.iter_content(chunk_size=self.CHUNK_SIZE):
-                fd.write(chunk)
+        existing_temporary_file = self._existing_tmp(target_file)
+
+        bytes_transfered = 0
+        if not existing_temporary_file:
+            mode = 'wb'
+            target_file = self.tmp_file(target_file)
```

Comment:
tmp_file creates a random file name. See:
```
return fname + '.' + str(uuid.uuid4())
```
So you won't be able to find your previous unfinished download.
Your test doesn't catch this bug because you patch return value of tmp_file and you shouldn't do that. You should use something like `fname + '.part'` instead of `self.tmp_file(fname)`.

Reply:
Actually, I think I will be able to find tmp file. See _existing_tmp(target_file):
```        for file in os.listdir(directory):
            targt_basename = os.path.basename(target_file)
            if targt_basename in file:
                return file```

I check if target_file is in name of file and, if so, it returns it. I have to admit that this is what I wanted to ask: Will such check be enough for us? It surely narrows usage of resume option.

Reply:
For the record: discussed this during our meeting.